components/usage/pkg/apiv1/billing.go

```diff
@@ -165,6 +177,61 @@ func (s *BillingService) CreateStripeCustomer(ctx context.Context, req *v1.Creat
 	}, nil
 }
 
+func (s *BillingService) CreateStripeSubscription(ctx context.Context, req *v1.CreateStripeSubscriptionRequest) (*v1.CreateStripeSubscriptionResponse, error) {
+	attributionID, err := db.ParseAttributionID(req.GetAttributionId())
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Invalid attribution ID %s", attributionID)
+	}
+
+	if req.CustomerId == "" {
+		return nil, status.Error(codes.InvalidArgument, "Invalid customer ID")
+	}
+
+	_, err = s.stripeClient.SetDefaultPaymentForCustomer(ctx, req.CustomerId, req.SetupIntentId)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to set default payment for customer ID %s", req.CustomerId)
```

Comment:
The return here is also the Status Code of the RPC we're returning. We should be explicit about the status.

```
log.Log.WithError(err).Error("...")
return nil, status.Errorf(codes.Internal, "...")
```

Reply:
And the same applies for the other returns.